src/index.ts

```diff
@@ -371,6 +377,178 @@ export const createWapeSaleInstance = (
+        throw Error(
+          `No claim could be found for user ${address} because they are not on the whitelist`
+        );
+      }
+      return userClaim;
+    },
+    getTotalForSale: async (): Promise<ethers.BigNumber> => {
+      const contract = await getGenSaleContract(
+        config.web3Provider,
+        config.contractAddress
+      );
+      const total = await contract.amountForSale();
+
+      return total;
+    },
+    getNumberOfDomainsSold: async (): Promise<ethers.BigNumber> => {
+      const contract = await getGenSaleContract(
+        config.web3Provider,
+        config.contractAddress
+      );
+      const domainsSold = await contract.domainsSold();
+      return domainsSold;
+    },
+    getBlockNumber: async (): Promise<number> => {
+      const blockNum = await config.web3Provider.getBlockNumber();
+      return blockNum;
+    },
+    getEthBalance: async (address: string): Promise<string> => {
+      const balance = await config.web3Provider.getBalance(address);
+      return ethers.utils.formatEther(balance);
+    },
+    isUserOnMintlist: async (address: string): Promise<boolean> => {
+      const mintlist = await getMintlist();
+      const isOnWhitelist = mintlist.claims[address] ? true : false;
+      return isOnWhitelist;
+    },
+    getDomainsPurchasedByAccount: async (address: string): Promise<number> => {
+      const contract = await getGenSaleContract(
+        config.web3Provider,
+        config.contractAddress
+      );
+      const domains = await contract.domainsPurchasedByAccount(address);
+      return domains.toNumber();
+    },
+    purchaseDomains: async (
+      count: ethers.BigNumber,
+      signer: ethers.Signer
+    ): Promise<ethers.ContractTransaction> => {
+      const contract = await getGenSaleContract(
+        signer,
+        config.contractAddress
+      );
+
+      const mintlist = await getMintlist();
+
+      const tx = await genSaleActions.purchaseDomains(
+        count,
+        signer,
+        contract,
+        mintlist
+      );
+      return tx;
+    },
+    setPauseStatus: async (
+      pauseStatus: boolean,
+      signer: ethers.Signer
+    ): Promise<ethers.ContractTransaction> => {
+      const contract = await getGenSaleContract(
+        signer,
+        config.contractAddress
+      );
+      const tx = await genSaleActions.setPauseStatus(
+        pauseStatus,
+        contract,
+        signer
+      );
+      return tx;
+    },
+    numberPurchasableByAccount: async (address: string): Promise<number> => {
+      const mintlist = await getMintlist();
+      const contract = await getGenSaleContract(
+        config.web3Provider,
+        config.contractAddress
+      );
+
+      const amount = await genSaleActions.numberPurchasableByAccount(
+        mintlist,
+        contract,
+        address
+      );
+
+      return amount;
+    },
+    setSalePrice: async function (price: ethers.BigNumber, signer: ethers.Signer): Promise<ethers.ContractTransaction> {
```

Comment:
This function is the same as the previous sales, and I dont believe our price variable changed in the Gen contract. It  should be the same as previous sales.

Reply:
I don't even know if we need this function in the SDK. If leadership wanted to change the configured price, or any other admin parameter they'd do it through us or the contract directly. Things like this aren't shown in the UI when doing sales for customers.

Reply:
Should I remove it?

Reply:
Your call. You didn't want to remove the generated typings and stuff I mentioned earlier because it would make this different from prior sales and so this would actually have an impact on the SDK (albeit negligible), and so the same argument could apply.

Reply:
Ive removed the method, I think it probably wont be used